File: solutions/0066-plus-one/solution.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let i = digits.length - 1;
    digits[i] += 1;
    
    while(digits[i] >= 10 && i >= 1){
        digits[i - 1] += 1;
        digits[i] = digits[i] % 10;
        i -= 1;
    }
    
    if(digits[0] >= 10){
        digits[0] = digits[0] % 10;
        digits.unshift(1);
    }
    
    return digits;
};

File: solutions/0066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        num = 0
        for digit in digits:
            num *= 10
            num += digit
        num += 1

        out = []
        while num > 0:
            out.append(num % 10)
            num //= 10

        out.reverse()
        return out
        
